src/main/java/be/ugent/rml/Executor.java

```diff
@@ -288,12 +309,17 @@ private List<PredicateObjectGraph> generatePredicateObjectGraphs(Mapping mapping
 
                 //check if we are dealing with a parentTriplesMap (RefObjMap)
             } else if (pogMapping.getParentTriplesMap() != null) {
-                List<ProvenancedTerm> objects;
+                Collection<ProvenancedTerm> objects;
```

Comment:
Is this change necessary? Does it improve performance?

Reply:
It is.

getIRIsWithSimpleJoin (new) handles a special case of joins much more efficiently (instead of comparing all pairs of tuples on a join using a predicate, it uses lookup with a map, making joins in the most common case – join on a single untransformed column in both tables – linear instead of quadratic).  getIRIsWithSimpleJoin returns a collection instead of a list to avoid copying the fields.

Yields linear improvement; in our cases, up to factor 1000-10000 for realistic data examples.

Since the use of objects does not depend on list-specific operations (random lookup) there is no downside to this even though the cost could be reduced in a preprocessing step.